pkg/engine/internal/workflow/workflow.go

```diff
@@ -33,14 +40,16 @@ type Workflow struct {
 
 	streamsMut   sync.RWMutex
 	streamStates map[*Stream]StreamState
+
+	admissionControl *admissionControl
 }
 
 // New creates a new Workflow from a physical plan. New returns an error if the
 // physical plan does not have exactly one root node, or if the physical plan
 // cannot be partitioned into a Workflow.
 //
 // The provided Runner will be used for Workflow execution.
-func New(logger log.Logger, tenantID string, runner Runner, plan *physical.Plan) (*Workflow, error) {
+func New(opts Options, logger log.Logger, tenantID string, runner Runner, plan *physical.Plan) (*Workflow, error) {
```

Comment:
smallest nit: the boundary for what's an option and what isn't seems unclear. Should all of these params be options? Or maybe at least Logger? 

(I don't mind if we don't change this right now, just something I was thinking about)